x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/types.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export type BoostType = 'value' | 'functional' | 'proximity';
+
+export interface Boost {
+  type: BoostType;
+  operation?: string;
+  function?: string;
+  newBoost?: boolean;
+  center?: string | number;
+  value?: string | number | string[] | number[];
+  factor: number;
+}
+
+export interface BoostObject {
+  [key: string]: Boost[];
+}
```

Comment:
[Not a change request, this is totally fine] I'm a Typescript nub and have no idea what I'm doing, but I wonder if `Record` is "technically" more correct for this type? e.g., `type BoostObject = Record<string, Boost[]>;`? 🤔 

Or maybe we don't need BoostObject at all (if we don't reuse it elsewhere) but instead just do
```ts
export interface SearchSettings {
  boosts: Record<string, Boost[]>;
  search_fields: object;
} 
```

Reply:
Oh that's neat, I didn't realize that existed.

Reply:
We have a lot of places that would be useful.

Reply:
f2db188

Reply:
Sweeeet! I'll keep an eye out for more Record opportunies as they come in